src/userobjects/OpenMCDomainFilterEditor.C

```diff
@@ -0,0 +1,206 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2024 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "OpenMCDomainFilterEditor.h"
+#include "OpenMCProblemBase.h"
+#include "openmc/tallies/tally.h"
+#include "openmc/tallies/filter.h"
+#include "openmc/tallies/filter_cell.h"
+#include "openmc/tallies/filter_material.h"
+#include "openmc/tallies/filter_universe.h"
+#include "openmc/tallies/filter_mesh.h"
+
+registerMooseObject("CardinalApp", OpenMCDomainFilterEditor);
+
+InputParameters
+OpenMCDomainFilterEditor::validParams()
+{
+  InputParameters params = OpenMCUserObject::validParams();
+  params.addParam<bool>("create_filter", false, "Whether to create the tally if it doesn't exist");
```

Comment:
Should this line say "filter" instead of "tally"?

Reply:
Yep! Addressed.